### PR TITLE
Use ReadCloser instead of Reader.

### PR DESCRIPTION
**Pull request by ajnavarro**

### CODE REVIEW

**Comment by guseggert** [comment]
This needs to close the reader at some point too right?

**Comment by ajnavarro** [comment]
Right, closing the reader using `defer` and improving harness test output to see the test output if they fail.

File: blueprints/base/services.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ var (
 )
 
 var (
-	IOReader           = &cg.GoRef{PkgPath: "io", Name: "Reader"}
+	IOReader           = &cg.GoRef{PkgPath: "io", Name: "ReadCloser"}
```
Comment (guseggert) [inline blueprints/base/services.go:27]:
```suggestion
	IOReadCloser           = &cg.GoRef{PkgPath: "io", Name: "ReadCloser"}
```

Comment (guseggert) [reply blueprints/base/services.go:27]: and also update https://github.com/ipld/edelweiss/blob/main/blueprints/services/dagjson-over-http/client.go#L58

**Review by petar — APPROVED** [review]
LGTM

**Pull request merged**